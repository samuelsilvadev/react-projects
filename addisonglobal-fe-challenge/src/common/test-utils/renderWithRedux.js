import React from 'react';
import { Provider } from 'react-redux';

import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';

export const renderWithRedux = ({ initialState, middlewares = [] }) => (
	component
) => {
	const mockStore = configureStore(middlewares);
	const store = mockStore(initialState);

	return render(<Provider store={store}>{component}</Provider>);
};
