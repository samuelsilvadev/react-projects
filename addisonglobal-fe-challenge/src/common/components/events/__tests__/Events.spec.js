import React from 'react';
import { render } from '@testing-library/react';

import ConnectedEvents, { Events } from '../Events';
import { renderWithRedux } from '../../../test-utils/renderWithRedux';

import eventsMock from './events.mock.json';

jest.mock('../../../store/actions/eventsActions', () => ({
	fetchEvents: () => ({ type: 'GET_EVENTS' })
}));

jest.mock('../../../contexts/AsideContext', () => ({
	useAsideContext() {
		return {
			handleOpen: () => {}
		};
	}
}));

jest.mock('../../market/Market', () => () => null);

describe('<Events />', () => {
	describe('Unconnected Events Component', () => {
		it('should call `fetchEvents` after components mount', () => {
			const props = {
				fetchEvents: jest.fn()
			};

			render(<Events {...props} />);

			expect(props.fetchEvents).toHaveBeenCalledTimes(1);
		});

		it('should render a list of events correctly', () => {
			const props = {
				fetchEvents: () => {},
				events: [
					{
						id: 1,
						name: 'Rick and Morty',
						markets: [
							{
								id: 1,
								name: 'first trip to mars'
							}
						]
					}
				]
			};

			const { container, queryByText } = render(<Events {...props} />);

			expect(queryByText('Rick and Morty')).toBeInTheDocument();
			expect(container.firstChild).toMatchSnapshot();
		});
	});

	describe('Connected Events', () => {
		it('should render correctly getting data from store', () => {
			const initialState = eventsMock;

			const { container, queryByText } = renderWithRedux({
				initialState
			})(<ConnectedEvents />);

			expect(queryByText('Real Madrid vs Barcelona')).toBeInTheDocument();
			expect(container.firstChild).toMatchSnapshot();
		});
	});
});
