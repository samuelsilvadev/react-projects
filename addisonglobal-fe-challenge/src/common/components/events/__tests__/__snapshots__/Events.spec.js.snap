// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Events /> Connected Events should render correctly getting data from store 1`] = `
<section
  class="event"
>
  <h3
    class="event__title"
  >
    Real Madrid vs Barcelona
  </h3>
  <ul
    class="event__market-wrapper"
  >
    <li
      class="event__market__item"
    />
  </ul>
</section>
`;

exports[`<Events /> Unconnected Events Component should render a list of events correctly 1`] = `
<section
  class="event"
>
  <h3
    class="event__title"
  >
    Rick and Morty
  </h3>
  <ul
    class="event__market-wrapper"
  >
    <li
      class="event__market__item"
    />
  </ul>
</section>
`;
